pkg/controller/util.go

```diff
@@ -326,3 +328,311 @@ func addToMap(m1, m2 map[string]string) map[string]string {
+					           },
+				        	},
+				        	TopologyKey: "Host-Assisted-Cloning",
+					    },
+					},
+				},
+			},
+			Containers: []v1.Container{
+				{
+					Command:		[]string{"/bin/sh"},
+					Name:            CLONER_SOURCE_PODNAME,
+					Image:           image,
+					ImagePullPolicy: v1.PullPolicy(pullPolicy),
+					SecurityContext: &v1.SecurityContext{
+						Privileged: &[]bool{true}[0],
+						RunAsUser: &[]int64{0}[0],
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      ImagePathName,
+							MountPath: CLONER_IMAGE_PATH,
+						},
+						{
+							Name:      SocketPathName,
+							MountPath: CLONER_SOCKET_PATH,
+						},
+					},
+					Args: []string{"-c", CLONER_SCRIPT_ARGS + " source"},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+			Volumes: []v1.Volume{
+				{
+					Name: ImagePathName,
+					VolumeSource: v1.VolumeSource{
+						PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+							ClaimName: pvcName,
+							ReadOnly:  false,
+						},
+					},
+				},
+				{
+					Name: SocketPathName,
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
+							Path: CLONER_SOCKET_PATH,
+						},
+					},
+				},
+			},
+		},
+	}
+	return pod
+}
+
+func CreateCloneTargetPod(client kubernetes.Interface, image string, verbose string, pullPolicy string, cr string, pvc *v1.PersistentVolumeClaim) (*v1.Pod, error) {
+	ns := pvc.Namespace
+    pod := MakeCloneTargetPodSpec(image, verbose, pullPolicy, cr, pvc)
+
+	pod, err := client.CoreV1().Pods(ns).Create(pod)
+	if err != nil {
+		return nil, errors.Wrap(err, "clone target pod API create errored")
+	}
+	glog.V(Vuser).Infof("cloning target pod \"%s/%s\" (image: %q) created\n", pod.Namespace, pod.Name, image)
+	return pod, nil
+}
+
+// return the clone target pod spec based on the target pvc.
+func MakeCloneTargetPodSpec(image, verbose, pullPolicy, cr string, pvc *v1.PersistentVolumeClaim) *v1.Pod {
+	// target pod name contains the pvc name
+	podName := fmt.Sprintf("%s-", CLONER_TARGET_PODNAME)
+
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: podName,
+			Annotations: map[string]string{
+				AnnCloningCreatedBy: "yes",
+			},
+			Labels: map[string]string{
+				"app": "Host-Assisted-Cloning",
+			},
+		},
+		Spec: v1.PodSpec{
+			Affinity: &v1.Affinity{
+				PodAffinity: &v1.PodAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+				        {
+				        	LabelSelector: &metav1.LabelSelector{
+					           MatchExpressions: []metav1.LabelSelectorRequirement{
+			                       {
+			              	          Key: "app",
+		                              Operator: metav1.LabelSelectorOpIn,
+							          Values: []string{"Host-Assisted-Cloning"},
+			                       },
+					           },
+				        	},
+				        	TopologyKey: "Host-Assisted-Cloning",
```

Comment:
`"Host-Assisted-Cloning"` is reused pretty often, consider making a constant in this func scope or in common.go